aodndata/moorings/products_handler.py

```diff
@@ -219,16 +230,16 @@ def _make_aggregated_timeseries(self):
             product_file.publish_type = PipelineFilePublishType.HARVEST_UPLOAD
             self.file_collection.add(product_file)
 
-            self._cleanup_previous_version(product_file.name, var)
+            self._cleanup_previous_version(product_file.name)
 
-    def _cleanup_previous_version(self, product_name, var):
-        """Delete any previously published version(s) of the product for this variable file.
+    def _cleanup_previous_version(self, product_name):
+        """Delete any previously published version(s) of the given product file.
         Ignores cases where the previous version has exactly the same file name, as this will simply be overwritten.
 
         :param product_name: Name of the newly generated product
-        :param var: Name of the variable of interest
         """
-        for old_product_url in self.old_product_files.get(var, []):
+        product_type = get_product_type(product_name)
+        for old_product_url in self.old_product_files.get(product_type, []):
             if os.path.basename(old_product_url) != product_name:
```

Comment:
IMO, the function intent is not clear by the directives within.
examples:

1.How ```old_product_url``` will ever match ```product_name```?  Maybe this difference if is better expressed with other 
names or the expression is better represented by a meaningful variable name ?

2.Also, I don't understand why we are ```self.file_collection.add(old_file)```  if we want to cleanup older ones. 

my point is, or the code above is not doing what it should be or the intent appears hidden in the implementation of other functions.

Reply:
I've updated the docstring and added a comment to explain this better.

Reply:
OK.

I still think the file_collection.add(old_file) is a bit anti-pattern because it leads the reader to think you are adding a file to a collection but you are in-fact scheduling the file to be removed later - so I would put more intent via a wrapper:

```
def sched_for_removal(self, file: str, **kwargs):
pfile = PipelineFile(file,**kwargs)
pfile.publish_type = ...
self.file_collection.add(pfile)
```
Also, the publish_type should be settable as kwargs too

Reply:
Yes, I _am_ adding a file to a collection. `self.file_collection` is the set of files the handler is operating on. Removing a file is one of the valid operations it can perform!

Anyway, I understand what you mean, but I think the comments I've added (plus an understanding of how `HandlerBase` works) should make it clear enough.